### PR TITLE
Add tests

### LINKED ISSUE #26: Chore: add unit tests and automated testing
Automate tests for PRs and configure checks.

### PR DESCRIPTION
**Pull request by satyarohith**
This PR introduces basic tests (which do not cover all edge cases yet) and also sets up github workflow to test on a pull request and a push to main.

Closes #26

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,10 +50,20 @@
...
   ],
+  "jest": {
+    "testMatch": [
+      "**/test/**/*.js",
+      "!**/test/fixtures/**/*.js"
+    ],
+    "verbose": true
+  },
   "devDependencies": {
+    "@netlify/build": "^5.0.1",
```
Comment (satyarohith) [inline package.json]: This isn’t required. I’ll update.

File: package.json [hunk 2]
```diff
@@ -46,10 +50,20 @@
   },
   "files": [
     "manifest.yml",
-    "index.js"
+    "src"
   ],
+  "jest": {
+    "testMatch": [
+      "**/test/**/*.js",
+      "!**/test/fixtures/**/*.js"
```
Comment (satyarohith) [inline package.json]: No longer needed. I’ll update.

**Review by rabbah — APPROVED** [review]
